### PR TITLE
Add generic modal component

### PR DESCRIPTION
**Pull request by mprasanjith**
## Description

- Refactor existing modal to fit the current UI design, and implement more generic behavior.
- Refactor existing modal hooks to be more generic.
- Refactor Terms of Sale modal to use new design.
- Add confirmation modal for token purchases in sales under soft cap.

## Motivation and Context

#261 

## How Has This Been Tested?

Tested locally using start-xdai. Purchase flow changes have been tested using token purchases.

## Screenshots (if appropriate):

image

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: src/views/Sale/components/PurchaseTokensForm/index.tsx [hunk 1]
```diff
@@ -166,50 +172,91 @@ export const PurchaseTokensForm = ({ saleId }: PurchaseTokensFormComponentProps)
...
-    [sale, library, account]
+    [toggleConfirmation, sale, totalBids, totalPurchased, purchaseTokens, purchaseValue]
+  )
+
+  const confirmationModalContent = (
+    <div>
+      <p>{t('texts.saleUnderSoftCap')}</p>
+
+      <p>
+        <Trans
```
Comment (rossneilson) [inline src/views/Sale/components/PurchaseTokensForm/index.tsx:247]: Is there a reason for using the Trans component over the useTranslation hook?
Interpolation should work with the hook:
https://www.i18next.com/translation-function/interpolation#basic

Comment (mprasanjith) [reply src/views/Sale/components/PurchaseTokensForm/index.tsx:247]: Oh, I somehow completely forgot this exists 🤦‍♂️ Fixing now.

Comment (rossneilson) [reply src/views/Sale/components/PurchaseTokensForm/index.tsx]: Nice, everything else looks good to me

**Review by rossneilson — APPROVED** [review]

**Review by martinkrung — CHANGES_REQUESTED** [review]
Working very nice, but two things

Screenshot at 2021-06-25 10-01-57

this results in the test not running without error, you also see this in the x here:

Screenshot at 2021-06-25 10-04-47


I think this is about the last commit, where you changed the way string works, but then forgot to remove the **Trans**

please change

And then:

After seeing the massage and move on, metamaks pops up, I decline the tx on metamask, then this happens

Screenshot at 2021-06-25 09-59-41

Maybe open a bugreport for this, if you think this has nothing to do with your changes.

**Comment by mprasanjith** [comment]
@martinkrung I've fixed the first issue. It indeed is about the last commit.

I stumbled upon the code causing the second issue. It's not something introduced by this change, but it's going to be a 5-minute fix, so I'll fix it in a separate PR.

**Comment by mprasanjith** [comment]
I ran into an issue where the popup was getting triggered even in the Approval process. (I'm assuming it doesn't make sense to show it while in approval flow, as its only applicable to purchases) It's related to #297, and should be fixed once that is fixed. I will put the PR back to ready state.

**Review by martinkrung — APPROVED** [review]
thanks, but we have go a minor merge conflict here: https://github.com/cryptonative-ch/mesa-interface/pull/292/conflicts

**Comment by mprasanjith** [comment]
I fixed the conflict. Merging the PR now.

**Pull request merged**